up-linux-streamer/examples/mE_service.rs

```diff
@@ -0,0 +1,110 @@
+    client: Arc<dyn UTransport>,
+}
+impl ServiceRequestResponder {
+    pub fn new(client: Arc<dyn UTransport>) -> Self {
+        Self { client }
+    }
+}
+#[async_trait]
+impl UListener for ServiceRequestResponder {
+    async fn on_receive(&self, msg: UMessage) {
+        println!("ServiceRequestResponder: Received a message: {msg:?}");
+
+        let Some(payload_bytes) = msg.payload else {
+            panic!("No bytes available");
+        };
+        let hello_request = match HelloRequest::parse_from_bytes(&payload_bytes) {
+            Ok(hello_request) => {
+                println!("hello_request: {hello_request:?}");
+                hello_request
+            }
+            Err(err) => {
+                error!("Unable to parse HelloRequest: {err:?}");
+                return;
+            }
+        };
+
+        let hello_response = HelloResponse {
+            message: format!("The response to the request: {}", hello_request.name),
+            ..Default::default()
+        };
+
+        let response_msg = UMessageBuilder::response_for_request(msg.attributes.as_ref().unwrap())
+            .build_with_protobuf_payload(&hello_response)
+            .unwrap();
+        self.client.send(response_msg).await.unwrap();
+    }
+
+    async fn on_error(&self, err: UStatus) {
+        println!("ServiceRequestResponder: Encountered an error: {err:?}");
+    }
+}
+
+#[tokio::main]
+async fn main() -> Result<(), UStatus> {
+    env_logger::init();
+
+    println!("mE_service");
+
+    let crate_dir = env!("CARGO_MANIFEST_DIR");
+    // TODO: Make configurable to pass the path to the vsomeip config as a command line argument
+    let vsomeip_config = PathBuf::from(crate_dir).join("vsomeip-configs/mE_service.json");
+    let vsomeip_config = canonicalize(vsomeip_config).ok();
+    trace!("vsomeip_config: {vsomeip_config:?}");
+
+    // There will be a single vsomeip_transport, as there is a connection into device and a streamer
+    // TODO: Add error handling if we fail to create a UPTransportVsomeip
+    let service: Arc<dyn UTransport> = Arc::new(
+        UPTransportVsomeip::new_with_config(
+            &SERVICE_AUTHORITY.to_string(),
+            SERVICE_UE_ID,
+            &vsomeip_config.unwrap(),
+        )
+        .unwrap(),
+    );
+
+    let source_filter = UUri {
+        authority_name: "*".to_string(),
+        ue_id: 0x0000_FFFF,
+        ue_version_major: 0xFF,
+        resource_id: 0xFFFF,
+        ..Default::default()
+    };
+    let sink_filter = UUri {
+        authority_name: SERVICE_AUTHORITY.to_string(),
+        ue_id: SERVICE_UE_ID as u32,
+        ue_version_major: SERVICE_UE_VERSION_MAJOR as u32,
+        resource_id: SERVICE_RESOURCE_ID as u32,
+        ..Default::default()
+    };
+    let service_request_responder: Arc<dyn UListener> =
+        Arc::new(ServiceRequestResponder::new(service.clone()));
+    // TODO: Need to revisit how the vsomeip config file is used in non point-to-point cases
+    service
```

Comment:
use `up_rust::communication::InMemoryRpcServer`

Reply:
I'll bring this in as a part of this PR: https://github.com/eclipse-uprotocol/up-streamer-rust/pull/26

Will mark this as resolved and to be tackled over there